Sources/SwiftInspectorVisitors/Tests/FunctionDeclarationVisitorSpec.swift

```diff
@@ -0,0 +1,123 @@
+import Foundation
+import Nimble
+import Quick
+import SwiftInspectorTestHelpers
+import SwiftInspectorVisitors
+
+final class FunctionDeclarationVisitorSpec: QuickSpec {
+  private var sut: FunctionDeclarationVisitor!
+
+  override func spec() {
+    beforeEach {
+      self.sut = FunctionDeclarationVisitor()
+    }
+
+    describe("visit(_:)") {
+      context("a function with a return value") {
+        beforeEach {
+          let content = """
+            func greet(person: String) -> String {
+                let greeting = "Hello, " + person + "!"
+                return greeting
+            }
+            """
+
+          try? self.sut.walkContent(content)
+        }
+
+        it("finds the function name") {
+          expect(self.sut.functionDeclarations.first?.name) == "greet"
+        }
+        it("finds the return type") {
+          expect(self.sut.functionDeclarations.first?.returnType) == TypeDescription.simple(name: "String")
+        }
+      }
+
+      context("a function with a return value") {
```

Comment:
this function does not have a return value

Reply:
copy-paste 😁